spring-cloud-gcp-autoconfigure/src/test/java/org/springframework/cloud/gcp/autoconfigure/secretmanager/it/SecretManagerIntegrationTests.java

```diff
@@ -0,0 +1,131 @@
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.secretmanager.it;
+
+import java.util.stream.StreamSupport;
+
+import com.google.cloud.secretmanager.v1beta1.AddSecretVersionRequest;
+import com.google.cloud.secretmanager.v1beta1.CreateSecretRequest;
+import com.google.cloud.secretmanager.v1beta1.ProjectName;
+import com.google.cloud.secretmanager.v1beta1.Replication;
+import com.google.cloud.secretmanager.v1beta1.Secret;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient;
+import com.google.cloud.secretmanager.v1beta1.SecretManagerServiceClient.ListSecretsPagedResponse;
+import com.google.cloud.secretmanager.v1beta1.SecretName;
+import com.google.cloud.secretmanager.v1beta1.SecretPayload;
+import com.google.protobuf.ByteString;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.springframework.boot.WebApplicationType;
+import org.springframework.boot.builder.SpringApplicationBuilder;
+import org.springframework.cloud.gcp.autoconfigure.core.GcpContextAutoConfiguration;
+import org.springframework.cloud.gcp.autoconfigure.secretmanager.GcpSecretManagerBootstrapConfiguration;
+import org.springframework.cloud.gcp.core.GcpProjectIdProvider;
+import org.springframework.context.ConfigurableApplicationContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assumptions.assumeThat;
+
+public class SecretManagerIntegrationTests {
+
+	private static final String TEST_SECRET_ID = "my-secret";
+
+	private ConfigurableApplicationContext context;
+
+	private GcpProjectIdProvider projectIdProvider;
+
+	private SecretManagerServiceClient client;
+
+	@BeforeClass
+	public static void prepare() {
+		assumeThat(System.getProperty("it.secretmanager"))
+				.as("Secret manager integration tests are disabled. "
+						+ "Please use '-Dit.secretmanager=true' to enable them.")
+				.isEqualTo("true");
+	}
+
+	@Before
+	public void setupSecretManager() {
+		this.context = new SpringApplicationBuilder()
+				.sources(GcpContextAutoConfiguration.class, GcpSecretManagerBootstrapConfiguration.class)
+				.web(WebApplicationType.NONE)
+				.properties("spring.cloud.gcp.secretmanager.enabled=true")
+				.run();
+
+		this.projectIdProvider = this.context.getBeanFactory().getBean(GcpProjectIdProvider.class);
+		this.client = this.context.getBeanFactory().getBean(SecretManagerServiceClient.class);
+
+		createSecret(TEST_SECRET_ID, "the secret data");
+	}
+
+	@After
+	public void close() {
+		if (this.context != null) {
+			this.context.close();
+		}
+	}
+
+	@Test
+	public void testConfiguration() {
+		assertThat(this.context.getEnvironment().getProperty("spring-cloud-gcp.secrets.my-secret"))
+				.isEqualTo("the secret data.");
+	}
+
+	/**
+	 * Creates the secret with the specified payload if the secret does not already exist.
+	 */
+	private void createSecret(String secretId, String payload) {
```

Comment:
This reminds me, do we want to create a `SecretManagerTemplate`? You would also put the secret reading in there.

Reply:
+1, and this would also go into the new module.